pkg/dns/plugin/nsmap/name.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nsmap
+
+import (
+	"github.com/miekg/dns"
+)
+
+// stringRewriter rewrites a string
+type stringRewriter interface {
```

Comment:
We don't need to define this interface here. You can embed a `*remapStringRewriter` in `nameRewriterResponseRule` and `valueRewriterResponseRule`.

https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Reply:
This is private code coming for CoreDNS, copied as is. I would rather leave it that way to not introduced bugs.

Reply:
If it's copied from another repo, please put it in the third_party dir

Reply:
Let me be more precise: this code is coming from the CoreDNS rewrite plugin and I've trimmed it down to just support exact rewriting. I didn't want to go to far in changing the original code but while I'm at it I can just trim it down even further and remove the interface

Reply:
Thanks. It still needs to be in third_party

Reply:
I would probably suggest including a link in a comment to the stackoverflow post, but that code does not need to be in third_party.

Everything coming from another repo must be identified as to where it originated from (ideally GitHub link w/specific commit), it must retain the original copyright header, and it needs to go in third_party.